SU-HSDO/suhumsci
# SHS-5941: Install CKEditor anchor_link plugin

# READY FOR REVIEW

## Summary
- Install the "Anchor link" module and add to selected text formats.
- Add custom plugin to modify the anchor link dialog and add help text and anchor name copy functionality.

## Need Review By (Date)
- Log into a [tugboat test site](https://pr1690-neuobkfqb6yflq3lq8hvgjmpozdgcvaf.tugboatqa.com/user/login).
- [Create a flexible page](https://pr1690-neuobkfqb6yflq3lq8hvgjmpozdgcvaf.tugboatqa.com/node/add/hs_basic_page). You should see a new "Anchor" button in the CKEditor of the default Text Area paragraph:
    <img width="146" alt="Screenshot 2025-02-25 at 5 14 36 PM" src="https://github.com/user-attachments/assets/0819d5d5-1896-471b-9002-c00a2dc5f35a" />
- Click on the button. You should see a dialog to add a new anchor.
    <img width="320" alt="Screenshot 2025-02-25 at 5 17 43 PM" src="https://github.com/user-attachments/assets/b4389e36-cce8-4b90-aaf1-65e2691b2959" />
- Confirm the following:
    - Dialog works as expected (you can create a new anchor).
    - Help text is present and correct.
    - The `#[anchor-name]` placeholder updates when you type into the text field.
    - You can copy the anchor name to the clipboard.
 - Test other paragraph types and content types that use a CKEditor field. Confirm that the "Anchor" button works as expected when present. Note that some CKEditor fields use a text format that doesn't have the "Anchor" button enabled (for example, the Postcard paragraph that uses the "Minimal HTML with Styles" format). For this cases, confirm that the editor loads and works as expected. You can see [the ticket](https://app.clickup.com/t/36718269/SHS-5941) to check which text formats have the "Anchor" button enabled.

## Urgency
medium

## Steps to Test
1. Create a Flexible Page
2. Confirm that CKEditor includes a new button in the toolbar for adding anchors (see image below)
3. Test the anchor functionality
    - Select some test, add an anchor with some name (for example `anchor-test`)
    - Add some more text and then a link, pointing to the anchor (`#anchor-test`)
    - Save the page and confirm that the anchor link works as expected. You can also inspect the page to confirm that the anchor markup is being added
4. Edit the page again and confirm that the rest of the functionality in CKEditor is unchanged and works as expected.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1209518760565337
